### PR TITLE
actions: add cylc-uiserver to build env

### PR DESCRIPTION
**Pull request by oliver-sanders**
Add cylc-uiserver to the build/test environment.

Required for #300

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -28,7 +28,10 @@ on:
         description: 'Cylc Flow GitHub release tag (must be present in cylc-docs)'
         required: true
       cylc-rose-tag:
-        description: 'Cylc Rose GitHub release tag'
+        description: 'cylc rose github release tag'
+        required: true
+      cylc-uis-tag:
+        description: 'cylc rose github release tag'
```
Comment (oliver-sanders) [inline .github/workflows/deploy.yml:34]:
```suggestion
        description: 'cylc uiserver github release tag'
```

File: .github/workflows/deploy.yml [hunk 2]
```diff
@@ -69,13 +72,15 @@ jobs:
...
           cylc_rose: ${{ github.event.inputs.cylc-rose-tag }}
+          cylc_uis: ${{ github.event.inputs.cylc-uis-tag }}
         run: |
           # NOTE: Install with [all] so we can import plugins which may
           #       have extra dependencies.
           pip install  \
             "cylc-flow[all] @ git+https://github.com/cylc/cylc-flow@${cylc_flow}" \
             "metomi-rose[all] @ git+https://github.com/metomi/rose@${meto_rose}" \
-            "cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@${cylc_rose}"
+            "cylc-rose[all] @ git+https://github.com/cylc/cylc-rose@${cylc_rose}" \
```
Comment (wxtim) [inline .github/workflows/deploy.yml:82]: I tried to test this, and get a dependency error:
```
ERROR: ResolutionImpossible: for help visit https://pip.pypa.io/en/latest/user_guide/#fixing-conflicting-dependencies
    cylc-flow[all] 8.0rc1.dev0 depends on cylc-flow 8.0rc1.dev0 (from git+https://github.com/cylc/cylc-flow@master)
    cylc-rose[all] 0.3.0 depends on cylc-flow==8.0b3.*
```

would it be reasonable to loosen the `cylc-rose` dependency at https://github.com/cylc/cylc-rose/blob/c9eef34a118913a0137ac08d5b4330ac38454d49/setup.py#L31

**Review by wxtim — CHANGES_REQUESTED** [review]
I don't think it works in its current form.

File: .github/workflows/deploy.yml (hunk 2 above)
Comment (oliver-sanders) [reply .github/workflows/deploy.yml:82]: No, this version is correct.

**Comment by oliver-sanders** [comment]
This will work *after* https://github.com/cylc/cylc-rose/pull/94

**Comment by wxtim** [comment]
> This will work _after_ [cylc/cylc-rose#94](https://github.com/cylc/cylc-rose/pull/94)

94 is merged; this is still failing.

**Review by datamel — APPROVED** [review]
Ah dependency mismatch but strange with 94 merged and a rerun hasn't fixed.

**Comment by oliver-sanders** [comment]
> 94 is merged; this is still failing.

Just needed to be re-triggered.

**Review by wxtim — APPROVED** [review]

**Pull request merged**